lib/bambora/v1/batch_payment_resource.rb

```diff
@@ -2,14 +2,48 @@
 
 module Bambora
   module V1
+    # Bank Electronic Funds Transfer (CAD) and Automatic Clearing House (USD)
     class BatchPaymentResource
-      # Summary: Bank Electronic Funds Transfer (CAD) and Automatic Clearing House (USD)
-      # Docs: https://dev.na.bambora.com/docs/guides/batch_payment/
-      #       https://dev.na.bambora.com/docs/references/batch_payment/
-      # Endpoint: https://api.na.bambora.com/v1/batchpayments
-      def initialize(client:, sub_path:)
+      attr_reader :client, :api_key, :sub_path
+
+      def initialize(client:, api_key:)
         @client = client
-        @sub_path = sub_path
+        @api_key = api_key
+        @sub_path = '/v1/batchpayments'
+      end
+
+      ##
+      # Post batch payment data.
+      #
+      # @example
+      #   batch_payment_resource.create(
+      #     [{
+      #       super_type: 'E',
+      #       transaction_type: 'D',
+      #       institution_number: 12345,
+      #       transit_number: 123,
+      #       account_number: 1223456789,
+      #       amount: 10000,
+      #       reference_nubmer: 1234,
+      #       reccipient_name: 'Hup Podling',
+      #       customer_code: '02355E2e58Bf488EAB4EaFAD7083dB6A',
+      #       dynamic_description: 'The Skeksis',
+      #     }],
+      #   )
+      #
+      # @param transactions [Array] of hashes with payment data.
+      # @param opts[:process_now] [Integer] optional. Defaults to 1.
+      # @param opts[:process_date] [String] optional. Must exclude +process_now+ or pass +{ process_now: 0 }+
+      #
+      # @see https://dev.na.bambora.com/docs/references/batch_payment/
+      # @see https://dev.na.bambora.com/docs/guides/batch_payment/
+      def create(transactions, opts = { process_now: 1 })
```

Comment:
It doesn't seem like we maintain any clear convention for using named parameters and nameless parameters. Based on current resource implementations, does it make sense to make `transactions` a named, keyword parameter?

Reply:
Key word arguments don't work when combined with an options hash because Ruby's underlying keyword implementation relies on a hash itself. The parser would then see `create(transactions: transactions, process_now: 1) as one hash, rather than two.

This style of `arg, options` is fairly common in Ruby, so even though we haven't done it yet in this project, I think it matches a lot of what you see in other places in Ruby land.